### PR TITLE
[feat]: [PIE-4639]: Added filePathCheck and getRepoUrlAndCheckForFileUniqueness for Input Sets also.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes: 
Same functionality is to be introduced fro Input-Set import also while importing from Git as that in pipeline import

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger messagecheck

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35856)

**Comment by senjucanon2** [comment]
trigger ti0

File: 878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java [hunk 1]
```diff
@@ -199,4 +199,12 @@ public String getRepoUrl(String accountIdentifier, String orgIdentifier, String
         .getRepoUrl(scope, gitEntityInfo.getRepoName(), gitEntityInfo.getConnectorRef(), Collections.emptyMap())
         .getRepoUrl();
   }
+  public void filePathCheck() {
```
Comment (namanvermah) [inline 878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java:202]: rename to something like check file path root folder

**Review by namanvermah — CHANGES_REQUESTED** [review]

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 11 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java` line 202 at r4](https://reviewable.io/reviews/harness/harness-core/35856#-N8SK0au434pRIhftCgL-r4-202:-N8SYB7T54qxqRL31Ho9:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5d025a34b55545059dbd8f66493e2e42d000c9cc/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java#L202)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

rename to something like check file path root folder
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 562 at r5](https://reviewable.io/reviews/harness/harness-core/35856#-N8SvBbw3Lc-WeR68zCd:-N8SvBbw3Lc-WeR68zCe:b-gp7t4l) ([raw file](https://github.com/harness/harness-core/blob/36c55caf4af6d61bd986b2f936864f9c81ca3a50/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L562)):*
> ```Java
> 
>   private boolean isAlreadyImported(String accountIdentifier, String repoURL, String filePath) {
>     Long totalInstancesOfYAML = pmsPipelineRepository.countFileInstances(accountIdentifier, repoURL, filePath);
> ```

this should be input set repository

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 562 at r5](https://reviewable.io/reviews/harness/harness-core/35856#-N8SvBbw3Lc-WeR68zCd:-N8T3f2eAYOaRFE_KdRI:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/36c55caf4af6d61bd986b2f936864f9c81ca3a50/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L562)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

this should be input set repository
</blockquote></details>

Done.

**Review by namanvermah — CHANGES_REQUESTED** [review]
> Reviewed 1 of 5 files at r1, 1 of 4 files at r3, 3 of 7 files at r4, 5 of 6 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 554 at r6](https://reviewable.io/reviews/harness/harness-core/35856#-N8TBUAK5pyprgOvVREK:-N8TBUAL8fErVyl35php:btyzdh8) ([raw file](https://github.com/harness/harness-core/blob/2fdedf57954fc88ee1457142ed69051b5d43423b/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L554)):*
> ```Java
>           gitEntityInfo.getFilePath());
>     } else if (inputSetRepository.checkIfInputSetExists(accountIdentifier, repoURL, gitEntityInfo.getFilePath())) {
>       String error = "The Requested YAML with Pipeline Id: " + pipelineIdentifier + ", RepoURl: " + repoURL
> ```

error message should have input set id not pipeline id

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 411 at r6](https://reviewable.io/reviews/harness/harness-core/35856#-N8TC1oVB-shaMLbbzmZ:-N8TC1oVB-shaMLbbzm_:b-147ico) ([raw file](https://github.com/harness/harness-core/blob/2fdedf57954fc88ee1457142ed69051b5d43423b/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L411)):*
> ```Java
>       InputSetImportRequestDTO inputSetImportRequestDTO, Boolean isForceImport) {
>     gitAwareEntityHelper.checkRootFolder();
>     getRepoUrlAndCheckForFileUniqueness(
> ```

repo url is not being used?

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 411 at r6](https://reviewable.io/reviews/harness/harness-core/35856#-N8TC1oVB-shaMLbbzmZ:-N8XNoXb5Mi6kavan7vd:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2fdedf57954fc88ee1457142ed69051b5d43423b/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L411)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

repo url is not being used?
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 554 at r6](https://reviewable.io/reviews/harness/harness-core/35856#-N8TBUAK5pyprgOvVREK:-N8XNpR45T7RJI6IF9RQ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2fdedf57954fc88ee1457142ed69051b5d43423b/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L554)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

error message should have input set id not pipeline id
</blockquote></details>

Done.

**Review by namanvermah — DISMISSED** [review]
> Reviewed 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java` line 201 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XTrAp5pMrmjSQ03A7:-N8XTrAp5pMrmjSQ03A8:b-22bpyh) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java#L201)):*
> ```Java
>         .getRepoUrl(scope, gitEntityInfo.getRepoName(), gitEntityInfo.getConnectorRef(), Collections.emptyMap())
>         .getRepoUrl();
>   }
> ```

spacing issue

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/test/java/io/harness/gitaware/helper/GitAwareEntityHelperTest.java` line 238 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XTwLOBSIxfHGAcHtG:-N8XTwLP1psJEki0Yh1P:b-njhinu) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/878-ng-common-utilities/src/test/java/io/harness/gitaware/helper/GitAwareEntityHelperTest.java#L238)):*
> ```Java
>     assertThatThrownBy(() -> gitAwareEntityHelper.updateEntityOnGit(noRepoName, data, scope))
>         .hasMessage("No repo name provided.");
>   }
> ```

check spacing

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetService.java` line 64 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XU1-h9cTJTnQYv5Cz:-N8XU1-h9cTJTnQYv5D-:bhrfch) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetService.java#L64)):*
> ```Java
>   InputSetEntity importInputSetFromRemote(String accountId, String orgIdentifier, String projectIdentifier,
>       String pipelineIdentifier, String inputSetIdentifier, InputSetImportRequestDTO inputSetImportRequestDTO,
>       Boolean isForceImport);
> ```

lets keep it boolean if there's no specific reason for it

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 545 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUJoiEikDVw_K9A_O:-N8XUJoiEikDVw_K9A_P:b-ap9r7q) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L545)):*
> ```Java
>   }
> 
>   public String getRepoUrlAndCheckForFileUniqueness(String accountIdentifier, String orgIdentifier,
> ```

why is it public?

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 546 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUN7f0Srrq04Dr4JC:-N8XUN7f0Srrq04Dr4JD:b-yf8p6l) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L546)):*
> ```Java
> 
>   public String getRepoUrlAndCheckForFileUniqueness(String accountIdentifier, String orgIdentifier,
>       String projectIdentifier, String inputSetIdentifier, Boolean isForceImport) {
> ```

boolean

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 553 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUxN10vyObqvBT9MC:-N8XUxN10vyObqvBT9MD:bexfp5b) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L553)):*
> ```Java
>       log.info("Importing YAML forcefully with InputSet Id: {}, RepoURl: {}, FilePath: {}", inputSetIdentifier, repoURL,
>           gitEntityInfo.getFilePath());
>     } else if (inputSetRepository.checkIfInputSetExists(accountIdentifier, repoURL, gitEntityInfo.getFilePath())) {
> ```

why are we checking if input set exists or not here?

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java` line 390 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XVxKbEML_hArd2qOg:-N8XVxKbEML_hArd2qOh:b-gdi8s2) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java#L390)):*
> ```Java
> 
>   @Override
>   public boolean checkIfInputSetExists(String accountId, String repoURL, String filePath) {
> ```

accountIdentifier?

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java` line 402 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XW0rcA2y87vj3HvTC:-N8XW0rcA2y87vj3HvTD:bd7wc55) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java#L402)):*
> ```Java
> 
>   @Override
>   public void setRepoUrlForSave(InputSetEntity entityToSave) {
> ```

why public?

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java` line 201 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XTrAp5pMrmjSQ03A7:-N8Xbdb91oikeztA0Tiw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareEntityHelper.java#L201)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

spacing issue
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/test/java/io/harness/gitaware/helper/GitAwareEntityHelperTest.java` line 238 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XTwLOBSIxfHGAcHtG:-N8XbeJzBd6Bh0Xfvi-F:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/878-ng-common-utilities/src/test/java/io/harness/gitaware/helper/GitAwareEntityHelperTest.java#L238)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

check spacing
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 545 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUJoiEikDVw_K9A_O:-N8Xbin-347rkGoSieUz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L545)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

why is it public?
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 546 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUN7f0Srrq04Dr4JC:-N8Xbjwf4_bGUl3q3-rh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L546)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

boolean
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java` line 390 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XVxKbEML_hArd2qOg:-N8Xbndz6sElanjuXyNt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java#L390)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

accountIdentifier?
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java` line 402 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XW0rcA2y87vj3HvTC:-N8Xbnut0I8RSwzBfvXa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/repositories/inputset/PMSInputSetRepositoryCustomImpl.java#L402)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

why public?
</blockquote></details>

Done.

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: 13 file(s) to review. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetService.java` line 64 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XU1-h9cTJTnQYv5Cz:-N8Xcqn30X3ppImf3JJv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetService.java#L64)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

lets keep it boolean if there's no specific reason for it
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java` line 553 at r7](https://reviewable.io/reviews/harness/harness-core/35856#-N8XUxN10vyObqvBT9MC:-N8XeFgO4eA1MtOaNhjv:b-agl4jc) ([raw file](https://github.com/harness/harness-core/blob/0b428314bcdad60b2b92bc24470d8e06c5242722/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/PMSInputSetServiceImpl.java#L553)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

why are we checking if input set exists or not here?
</blockquote></details>

I want to check on the name. It suggests a generic input set exists method. Can we make it a bit more explicit, eg: something like "checkIfInputSetExistsForGivenFilePath" etc.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 4 files at r3, 3 of 7 files at r4, 3 of 6 files at r5, 2 of 5 files at r8, 1 of 2 files at r9, 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856#-:-N8XtEhY-y3ati6SRNu_:bnfp4nl)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r3, 3 of 7 files at r4, 3 of 6 files at r5, 2 of 5 files at r8, 1 of 2 files at r9, 3 of 3 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35856)* status: :shipit: complete! ✓

**Pull request merged**